airflow/providers/exasol/hooks/exasol.py

```diff
@@ -0,0 +1,179 @@
+                return cur.fetchall()
+
+    def get_first(self, sql, parameters=None):
+        """
+        Executes the sql and returns the first resulting row.
+
+        :param sql: the sql statement to be executed (str) or a list of
+            sql statements to execute
+        :type sql: str or list
+        :param parameters: The parameters to render the SQL query with.
+        :type parameters: mapping or iterable
+        """
+        with closing(self.get_conn()) as conn:
+            with closing(conn.execute(sql, parameters)) as cur:
+                return cur.fetchone()
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Runs a command or a list of commands. Pass a list of sql
+        statements to the sql parameter to get them to execute
+        sequentially
+
+        :param sql: the sql statement to be executed (str) or a list of
+            sql statements to execute
+        :type sql: str or list
+        :param autocommit: What to set the connection's autocommit setting to
+            before executing the query.
+        :type autocommit: bool
+        :param parameters: The parameters to render the SQL query with.
+        :type parameters: mapping or iterable
+        """
+        if isinstance(sql, basestring):
+            sql = [sql]
+
+        with closing(self.get_conn()) as conn:
+            if self.supports_autocommit:
+                self.set_autocommit(conn, autocommit)
+
+            for query in sql:
+                self.log.info(query)
+                with closing(conn.execute(query, parameters)) as cur:
+                    self.log.info(cur.row_count)
+            # If autocommit was set to False for db that supports autocommit,
+            # or if db does not supports autocommit, we do a manual commit.
+            if not self.get_autocommit(conn):
+                conn.commit()
+
+    def set_autocommit(self, conn, autocommit):
+        """
+        Sets the autocommit flag on the connection
+
+        :param conn: Connection to set autocommit setting to.
+        :type conn: connection object
+        :param autocommit: The autocommit setting to set.
+        :type autocommit: bool
+        """
+        if not self.supports_autocommit and autocommit:
+            self.log.warning(
+                ("%s connection doesn't support "
+                 "autocommit but autocommit activated."),
+                getattr(self, self.conn_name_attr))
+        conn.set_autocommit(autocommit)
+
+    def get_autocommit(self, conn):
+        """
+        Get autocommit setting for the provided connection.
+        Return True if autocommit is set.
+        Return False if autocommit is not set or set to False or conn
+        does not support autocommit.
+
+        :param conn: Connection to get autocommit setting from.
+        :type conn: connection object
+        :return: connection autocommit setting.
+        :rtype bool
```

Comment:
```suggestion
        :rtype: bool
```

Reply:
Thanks! I looked over the file some many times...

Reply:
Do you have IDE? Syntax coloring makes it very easy to find a bug.
![Screenshot 2020-03-06 at 16 00 21](https://user-images.githubusercontent.com/12058428/76094857-9ae23980-5fc3-11ea-95c9-8afd1f25bae4.png)

Reply:
I do, but clearly not one that highlights rst at this level. Something to look into.